### ISSUE TITLE
undefined local variable or method failure_message_args_description

### ISSUE THREAD

**Issue by olleolleolle** [splitwise/super_diff#113]
Hi!

My Capybara matcher (have_text) ran into this, when superdiff was in the `Gemfile`:

```
     NameError:
       undefined local variable or method `failure_message_args_description' for #<RSpec::Matchers::BuiltIn::Has:0x00005612812d0cc8>
       Did you mean?  failure_message_expecting
```

The method is called in the `monkey_patches.rb` file in this project. 

I expected to get a failure like this reported (which it does when I removed super_diff):

```
       expected `#<Capybara::Node::Simple tag="div" path="/html/body/div[1]">.has_content?("Receipt of co
llection")` to be truthy, got false
```

My Gemfile was using latest code in this repo:

```
GIT
  remote: https://github.com/mcmire/super_diff.git
  revision: 034c90225aeb59c10a7ee2898efedf8c4bc27e88
  specs:
    super_diff (0.5.2)
      attr_extras (>= 6.2.4)
      diff-lcs
      patience_diff
```

I searched in the rspec-expectations repo, too, and was unable to locate a method like that, by searching and looking. https://github.com/rspec/rspec-expectations

**Comment by mcmire** [comment]
Hmmm! I can't think of why exactly this is happening, but let me look into it and get back to you.

**Comment by olleolleolle** [comment]
Perhaps because a Capybara `Base` Matcher is not implementing all of the (private) interface?

https://github.com/teamcapybara/capybara/blob/8510765c68913ccd3ce8e4d89cefe514ee187087/lib/capybara/rspec/matchers/base.rb#L13

**Comment by mcmire** [comment]
@olleolleolle No... I think this might be happening due to a change to RSpec itself. `failure_message_args_description` is a private method in the Has matcher that seems to have been refactored out [here](https://github.com/rspec/rspec-expectations/commit/c028a9ae062a0262b33b8169631dc2d82be90715#diff-d3b3086ce048c1abcc8e9452f21c7afa019704480ba45fcf09c98819bcccdbc4) and [here](https://github.com/rspec/rspec-expectations/commit/e435a1db5a754a2fc8b458868a96d25c4f689c83#diff-d3b3086ce048c1abcc8e9452f21c7afa019704480ba45fcf09c98819bcccdbc4). It looks like the way that arguments are displayed comes from [here](https://github.com/rspec/rspec-expectations/blob/a09decea6e95e20c6374046ac5bb79128ab56fe9/lib/rspec/matchers/built_in/be.rb#L78) now. In fact, I just tried upgrading to RSpec 3.10 and re-running `spec/integration/rspec/have_predicate_spec.rb` locally and those tests fail. So... that's what's happening.

**Comment by arturopie** [comment]
I'm running into the same issue. Let me know if you need more info.

**Comment by mcmire** [comment]
I recently released 0.6.1 which contains a fix for this issue. So you shouldn't see this anymore, but if you do let me know!
